modules/default-flatpaks/v1/default-flatpaks.sh

```diff
@@ -17,26 +17,26 @@ configure_flatpak_repo () {
     CONFIG_FILE=$1
     INSTALL_LEVEL=$2
     REPO_INFO="/usr/share/bluebuild/default-flatpaks/$INSTALL_LEVEL/repo-info.yml"
-    get_yaml_array INSTALL ".$INSTALL_LEVEL.install[]" "$CONFIG_FILE"
+    get_json_array INSTALL "try .["$INSTALL_LEVEL.install"][]" "$CONFIG_FILE"
 
 
     # Checks pre-configured repo info, if exists
     if [[ -f $REPO_INFO ]]; then
         echo "Existing $INSTALL_LEVEL configuration found:"
         cat $REPO_INFO
-        CONFIG_URL=$(yq ".repo-url" "$REPO_INFO")
-        CONFIG_NAME=$(yq ".repo-name" "$REPO_INFO")
-        CONFIG_TITLE=$(yq ".repo-title" "$REPO_INFO")
+        CONFIG_URL=$(jq -r 'try .["repo-url"]' "$REPO_INFO")
+        CONFIG_NAME=$(jq -r 'try .["repo-name"]' "$REPO_INFO")
+        CONFIG_TITLE=$(jq -r 'try .["repo-title"]' "$REPO_INFO")
     else
         CONFIG_URL="null"
         CONFIG_NAME="null"
         CONFIG_TITLE="null"
     fi
 
     echo "Configuring $INSTALL_LEVEL repo in $REPO_INFO"
-    REPO_URL=$(echo "$CONFIG_FILE" | yq -I=0 ".$INSTALL_LEVEL.repo-url")
-    REPO_NAME=$(echo "$CONFIG_FILE" | yq -I=0 ".$INSTALL_LEVEL.repo-name")
-    REPO_TITLE=$(echo "$CONFIG_FILE" | yq -I=0 ".$INSTALL_LEVEL.repo-title")
+    REPO_URL=$(echo "$CONFIG_FILE" | jq -r 'try .["$INSTALL_LEVEL.repo-url"]')
```

Comment:
The variable expansion is in single quotes, which means it won't be populated properly.

Reply:
Solved